jetson/percep/pcl.cpp

```diff
@@ -186,177 +188,81 @@ void PCL::FindInterestPoints(std::vector<pcl::PointIndices> &cluster_indices,
 
 }
 
+/* --- Get Angle Off Center--- */
+//This function finds the angle off center the
+//line that passes through both these points is
+//Direction of 0 is left and 1 is right
+double PCL::getAngleOffCenter(int buffer, int direction, std::vector<std::vector<int>> &interest_points,
+                    shared_ptr<pcl::visualization::PCLVisualizer> viewer, std::vector<int> &obstacles){
+    double newAngle = 0;
+    //If Center Path is blocked check the left or right path depending on direction parameter
+    while(newAngle > -70 && newAngle < 70) {
+        
+        //Finding angle off center
+        double xDist = pt_cloud_ptr->points[obstacles.at(direction)].x;
+        double zDist = pt_cloud_ptr->points[obstacles.at(0)].z;//Length of adjacent
+        xDist       += direction ? buffer+HALF_ROVER : -(buffer+HALF_ROVER); //Calculate length of opposite
+        newAngle     = atan(xDist/zDist)*180/PI;//arctan(opposite/adjacent)
+
+        //Create compareLine Functors
+        compareLine leftLine(newAngle, -HALF_ROVER);
+        compareLine rightLine(newAngle, HALF_ROVER);
+        
+        obstacles.clear();
+        
+        if(CheckPath(interest_points, viewer, obstacles, leftLine, rightLine)) {
+            #if PERCEPTION_DEBUG
+                std::cout<<"!!!!!!!!!!!!!!!!!!!!!!!!!!!!!FOUND NEW PATH AT: "<<newAngle<<std::endl;
+            #endif
+            return newAngle;
+        }    
+    }
+    return 360;
+}
+
 /* --- Find Clear Path --- */
 //Returns the angle to a clear path
-double PCL::FindClearPath(std::vector<std::vector<int>> interest_points,
-                        shared_ptr<pcl::visualization::PCLVisualizer> viewer) {
+double PCL::FindClearPath(std::vector<std::vector<int>> &interest_points,
+                        shared_ptr<pcl::visualization::PCLVisualizer> viewer) {                        
+    
     #if PERCEPTION_DEBUG
         pcl::ScopeTime t ("Find Clear Path");
     #endif
 
-    int buffer = 50;
-
-    std::vector<int> obstacles; //X and Y values of the left and rightmost obstacles in path
+    // create left and right angle off center that default at 360 degrees
+    double leftAngle  = 360;
+    double rightAngle = 360;
```

Comment:
Same question as above, why 360?

Reply:
It's just an outlier that we can check for since we only send values between -70 and 70

Reply:
What happens when nav receives 360? This seems like something we don't want to have happen. Even if it _shouldn't_ happen, there will be a case when it does happen.

Reply:
When nav recieves 360 that means there's no clear path and we've probably run into a wall or something. Does nav have a state for that?